trino/client.py

```diff
@@ -127,6 +138,9 @@ def __init__(
         self._roles = roles.copy() if roles is not None else {}
         self._prepared_statements: Dict[str, str] = {}
         self._object_lock = threading.Lock()
+        if timezone:  # Check timezone validity
+            ZoneInfo(timezone)
+        self._timezone = timezone or get_localzone_name()
```

Comment:
`get_localzone_name` can also return invalid zone names.

IMO the `ZoneInfo` check should be moved after this line as `ZoneInfo(self._timezone)`

Reply:
Good catch. Thanks